---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# RASC

<!-- badges: start -->
<!-- badges: end -->

`RASC` is a tool that reproduce and improve the workflow of the single cell RNA analysis which includes processing and filtering the data, calculating dimension reduction, unsupervised clustering,and cell type annotation. The goal of scRNA analysis is to do clustering and classifying cells at single cell level. It will perform analysis on the gene expression in every cell with a reference, then annotating the corresponding cell types.The reference should includes the unique gene expression of different cell type. The improvement would be done in the labeling of cell type part. For existing
scRNA analysis tool such as Deseq2, they labelled the cell types based on reference signature and the score of each reference gene based on the similarity between sample cells and reference cell’s gene expression. For example, they will label the cell as type A because this cell mostly resemble gene expression of cell A. Yet, in some circumstances, some cells don’t resemble any cells, so they will have a very low score for every reference cell types. However, due to the property of the algorithm, they will still be labeled with highest score of reference cells they resemble. Thus, some weird cells might be filtered or marked as normal cells. My package will fix this problem by provide alternative way to label the cell type.My package will rank the cell based on their enrichment in different pathways,so users might then use this information to label the cell other than compare with the reference cell signature to make the labeling part more accurate.
The version of R is 4.1.0 and the platform used for this package is windows

## Installation

You can install the development version of RASC like so:

```R
require("devtools")
devtools::install_github("MengyiShu/RASC", build_vignettes = TRUE)
library("RASC")
```
To run the shinyApp:
``` r
runTestingPackage()
```

## Overview
```R
ls("package:RASC")
data(package = "RASC")
```
An overview of the package is illustrated below:
![](./inst/extdata/RASC.png)

```R
browseVignettes("IntroductionToRASC")
```

## Reference
Lawrence, M., & Turaga, N. (2021, March 27). Orchestrating Single Cell Analysis. GitHub. Retrieved
October 26, 2021, from https://github.com/Bioconductor/ OrchestratingSingleCellAnalysis

## Acknowledgements
This package was developed as part of an assessment for 2021 BCB410H: Applied Bioinformatics, University of Toronto, Toronto, CANADA.
